pandas/io/tests/test_parsers.py

```diff
@@ -2664,6 +2664,35 @@ def test_empty_header_read(count):
         for count in range(1, 101):
             test_empty_header_read(count)
 
+    def test_read_empty_with_usecols(self):
+        # See gh-12493
+        names = ['Dummy', 'X', 'Dummy_2']
+        usecols = names[1:2] # ['X']
+
+        def check_usecols(stringIO, expected):
+            df = self.read_csv(stringIO, names=names, usecols=usecols)
+            tm.assert_frame_equal(df, expected)
+
+        # first, check to see that the response of
+        # parser when faced with no provided columns
+        # throws the correct error, with or without usecols
+        name = self.__class__.__name__
+        errmsg = "No columns to parse from file"
+        klass = (ValueError if 'CParser' in name else StopIteration)
```

Comment:
Actually, scrap that idea of mine.  That causes a lot of test failures when I do that.